tests/python/pants_test/test_base.py

```diff
@@ -698,6 +699,18 @@ def captured_logging(self, level=None):
       root_logger.setLevel(old_level)
       root_logger.removeHandler(handler)
 
+  @contextmanager
+  def warnings_catcher(self):
+    with warnings.catch_warnings(record=True) as w:
+      warnings.simplefilter('always')
+      yield w
+
+  def assertWarning(self, w, category, warning_text):
+    single_warning = assert_single_element(w)
+    self.assertEqual(single_warning.category, category)
+    warning_message = single_warning.message
+    self.assertEqual(warning_text, text_type(warning_message))
```

Comment:
Ditto. Do you anticipate this being used beyond the deprecation test files?

Reply:
This could perhaps be moved to `test_deprecated.py`, but the code is rather fiddly and if someone else wanted to catch a warning message, I would prefer they look at using this method which checks for an exact string match and checks by category instead of doing something more coarse-grained (like just asserting it contains a string) due to lack of documentation.

Reply:
This was moved from `test_options.py`, which now has `self.assertOptionWarning()` for its specific needs.